src/lib/class/Server/methods.cpp

```diff
@@ -0,0 +1,186 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   methods.cpp                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: jodufour <jodufour@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/03/03 22:06:33 by jodufour          #+#    #+#             */
+/*   Updated: 2024/03/06 02:23:41 by jodufour         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "class/Message.hpp"
+#include "class/Server.hpp"
+#include "class/exception/ProblemWithAccept.hpp"
+#include "class/exception/ProblemWithEpollCtl.hpp"
+#include "class/exception/ProblemWithEpollWait.hpp"
+#include "class/exception/ProblemWithFcntl.hpp"
+#include "class/exception/ProblemWithRecv.hpp"
+#include "replies.hpp"
+#include <fcntl.h>
+
+bool interrupted = false;
+
+/**
+ * @brief Starts the server, making it receiving messages, handling them and sending the appropriate replies.
+ */
+void Server::start(void)
```

Comment:
I tend to think that a function calling other functions should be after theses functions in the code

Reply:
We can do that if you want to. I used to define the methods in the same order as they appear in the header file, but I also used to define the sub-functions before the functions that call them outside of classes. So I don't mind doing the same for the methods.

Reply:
If you defined all of them in the same order as the header, we can keep it like that if everything is coherent.